metadata-ingestion/src/datahub/ingestion/source/mlflow.py

```diff
@@ -262,6 +264,92 @@ def _convert_run_result_type(
                 type="SKIPPED", nativeResultType=self.platform
             )
 
+    def _get_dataset_schema(self, schema: str) -> Optional[List[Tuple[str, str]]]:
+        try:
+            schema_dict = json.loads(schema)
+        except json.JSONDecodeError:
+            print("Failed to parse schema JSON")
+            return None
+
+        if "mlflow_colspec" in schema_dict:
+            try:
+                return [
+                    (field["name"], field["type"])
+                    for field in schema_dict["mlflow_colspec"]
+                ]
+            except (KeyError, TypeError):
+                return None
+        # If the schema is not formatted, return None
+        return None
+
+    def _get_dataset_platform_from_source_type(self, source_type):
+        # manually map mlflow platform to datahub platform
+        if source_type == "gs":
+            return "gcs"
+        return source_type
+
+    def _get_dataset_input_workunits(self, run: Run) -> Iterable[MetadataWorkUnit]:
+        run_urn = DataProcessInstance(
+            id=run.info.run_id,
+            orchestrator=self.platform,
+        ).urn
+        dataset_inputs = run.inputs.dataset_inputs
+        dataset_reference_urns = []
+        for dataset_input in dataset_inputs:
+            source_type = dataset_input.dataset.source_type
+            dataset_tags = {k[1]: v[1] for k, v in dataset_input.tags}
+            dataset = dataset_input.dataset
+            platform = self._get_dataset_platform_from_source_type(source_type)
```

Comment:
looks like this method is being called twice

Reply:
nice catch 🫠 done.